cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerServiceTest.java

```diff
@@ -151,4 +151,47 @@ private void validateUnpackDir(File unpackedFile) {
     Assert.assertTrue(files.size() > 1);
     Assert.assertTrue(files.stream().anyMatch(s -> s.equals("META-INF")));
   }
+
+  @Test
+  public void testArtifact() throws Exception {
```

Comment:
Should we also check that the returned location is a packed file.

Reply:
Done